### PR TITLE
feature: when `stripBasePath` is set to false, do not strip basePath

### PR DESCRIPTION
**Pull request by andyburke**
Related to #232 #233

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: dynamic.js [hunk 1]
```diff
@@ -134,7 +134,7 @@ module.exports = function (fastify, opts, next) {
       const schema = transform
         ? transform(route.schema)
         : route.schema
-      let path = route.url.startsWith(basePath)
+      let path = route.url.startsWith(basePath) && !process.env.FASTIFY_SWAGGER_LEAVE_BASE_PATH
```
Comment (mcollina) [inline dynamic.js:137]: Could you move this behind a plugin a plugin option?

**Review by mcollina — CHANGES_REQUESTED** [review]

File: dynamic.js (hunk 1 above)
Comment (andyburke) [reply dynamic.js:137]: I'm not sure I understand. Are you asking to make this into a plugin or to make this controlled by a passed-in option?

Comment (andyburke) [reply dynamic.js]: I've made some changes that I hope address what you were requesting. Please let me know what you think.

File: index.d.ts [hunk 2]
```diff
@@ -46,6 +46,11 @@ export interface FastifyDynamicSwaggerOptions extends FastifySwaggerOptions {
   mode?: 'dynamic';
   swagger?: Partial<SwaggerSchema.Spec>;
   hiddenTag?: string;
+  /**
+   * Strips matching base path from routes in documentation
+   * @default true
+   */
+  stripBasePath?: boolean;
```
Comment (mcollina) [inline index.d.ts:53]: We also have a test for the types, can you add this option there as well? Thanks

**Review by mcollina — COMMENTED** [review]
good work! Just a minor thing

File: index.d.ts (hunk 2 above)
Comment (andyburke) [reply index.d.ts:53]: I pushed a small addition to the tests, but I'm not sure if it's what you were asking for.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**